### PR TITLE
[Merged by Bors] - feat: port RingTheory.QuotientNilpotent

### PR DESCRIPTION
**Pull request by mattrobball**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/QuotientNilpotent.lean [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  · rw [pow_zero, Ideal.one_eq_top] at hI
+    haveI := subsingleton_of_bot_eq_top hI.symm
+    exact (hI' (Subsingleton.elim _ _)).elim
+  cases' n with n
+  · rw [pow_one] at hI
+    exact (hI' hI).elim
+  apply h₂ (I ^ 2) _ (Ideal.pow_le_self two_ne_zero)
+  · apply H n.succ _ (I ^ 2)
+    · rw [← pow_mul, eq_bot_iff, ← hI, Nat.succ_eq_add_one, Nat.succ_eq_add_one]
+      -- Porting note: linarith wants AddGroup (Ideal S) to solve (n:ℕ)+1+1 ≤ 2*(n+1) 🤯
```
Comment (Parcly-Taxel) [inline Mathlib/RingTheory/QuotientNilpotent.lean:64]:
```suggestion
      -- Porting note: linarith wants AddGroup (Ideal S) to solve (n:ℕ)+1+1 ≤ 2*(n+1)
```

**Review by Parcly-Taxel — APPROVED** [review]

**Comment by ChrisHughes24** [comment]
bore merge

**Comment by Parcly-Taxel** [comment]
> bore merge

@ChrisHughes24 "bore"?

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**